python/taichi/misc/gui.py

```diff
@@ -139,24 +139,45 @@ def show(self, file=None):
             self.core.screenshot(file)
         self.clear(self.background_color)
 
+    class EventFilter:
+        def __init__(self, *filter):
+            self.filter = set(filter)
+
+        def match(self, e):
+            if (e.type, e.key) in self.filter:
+                return True
+            if e.type in self.filter:
+                return True
+            if e.key in self.filter:
+                return True
+            return False
+
     def has_key_event(self):
         return self.core.has_key_event()
 
     def get_event(self, *filter):
+        for e in self.get_events(*filter):
+            self.event = e
+            return True
+        return False
+
+    def get_events(self, *filter):
+        filter = filter and GUI.EventFilter(*filter) or None
```

Comment:
I know I suggested `filter or None`, but in this case, i think the idiomatic way is `GUI.EventFilter(*filter) if filter else None`.